### PR TITLE
Added textbox, -v, -h, @build/generic, @build/c, @build/cpp, @run/node, Updated logging, added top-level config, createTimeTracker, TextboxBuilder, createTextbox

### PR DESCRIPTION
**Pull request by FishingHacks**

### CODE REVIEW

**Review by RedCrafter07 — DISMISSED** [review]
I'd remove examples/src/test.ts, as it's a test file. Test files are not needed in production.

> I've checked most of the files. The new module files will have a detailed "look over" in a few hours.

**Comment by RedCrafter07** [comment]
Also, small comment for the future: You may add a description and not such long titles (161 Chars) 😅

**Comment by FishingHacks** [comment]
main.ts (formerly test.ts) is required, as it would make no sense to have a ts-compilation example without typescript files

**Comment by RedCrafter07** [comment]
Oh wait, haven't seen the path

**Comment by RedCrafter07** [comment]
It was this morning 😅

File: src/modules/@init/typescript.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { sync } from 'cross-spawn';
+import { is } from '../../lib/utils';
+import { Module } from '../../types';
+
+export default {
+    validate: () => true,
+    initiate: () => {
+        if (is.processError(sync('tsc', ['--init'])))
+            console.error(chalk.redBright('[!] TSC is not installed'));
+        else console.log(chalk.greenBright('[+] Installed chalk'));
```
Comment (RedCrafter07) [inline src/modules/@init/typescript.ts:11]: Why is there an "Install chalk"? 🤔

Comment (FishingHacks) [reply src/modules/@init/typescript.ts:11]: wus

Comment (FishingHacks) [reply src/modules/@init/typescript.ts:11]: wie ist das da hingekommen

Comment (RedCrafter07) [reply src/modules/@init/typescript.ts:11]: In english please 😂

Comment (RedCrafter07) [reply src/modules/@init/typescript.ts]: I'll mark this as resolved.

**Review by RedCrafter07 — APPROVED** [review]
Great work, Fishi 👍

**Comment by RedCrafter07** [comment]
I'll test it soon and merge it afterwards.

**Comment by FishingHacks** [comment]
# Instructions

-v - run redstart -v
-h - run redstart -h
-u - run redstart -u <modulename>
-m - run redstart -m 

@build/generic:
```yaml
@build/generic

[@build/generic]
command: echo
arguments: hello world
```
Should print hello world to the output

@build/c and @build/cpp
Make a project with .c or .cpp files
```yaml
@build/c

[@build/c]
fileName: out
sourceDirectory: .
optimizations: 3
```

@run/node
a project with a index.js file
```yaml
@run/node


[@run/node]
mainFile: index.js
```

updated logging & added top-level config & createTimeTracker:
Add `--dbgprint: true` to any file at the top-level

TextboxBuilder:
run redstart -h

createTextbox:
unused

template system:
run redstart -t

exitonerr tlc:
Add `--dbgprint: true` to any erroring .rsproj file. See examples/test.rsproj. Expected: No log with Hi!

@init/typescript:
```yaml
@init/typescript

[@init/typescript]
```

@build/typescript:
A project with .ts files, dist is out file
```yaml
@build/typescript

[@build/typescript]
allowJSFiles: true
buildDirectory: dist
```

File: src/bin/main.ts [hunk 2]
```diff
@@ -1,63 +1,230 @@
...
  * @license GPL3
- * @author RedCrafter07 (https://github.com/RedCrafter07)
+ * @author RedCrafter07 <https://github.com/RedCrafter07>
+ * @author FishingHacks <https://github.com/FishingHacks>
  */
 
 import chalk from 'chalk';
-import { existsSync, lstatSync } from "fs";
-import { readdir } from 'fs/promises';
+import { existsSync, lstatSync, readdirSync, readFileSync } from 'fs';
```
Comment (RedCrafter07) [inline src/bin/main.ts:9]: Why do you need readdirSync and readFileSync? readdir & readfile from fs/promises to the same.

Comment (FishingHacks) [reply src/bin/main.ts:9]: Well, i can change it in the next pr, have a system half-ready and don't want to remove it

Comment (RedCrafter07) [reply src/bin/main.ts:9]: Okay

**Comment by RedCrafter07** [comment]
Merge incoming...

**Pull request merged**